java/flight/flight-core/src/main/java/org/apache/arrow/flight/auth2/GeneratedBearerTokenAuthHandler.java

```diff
@@ -44,7 +44,7 @@ public boolean validateBearer(String bearerToken) {
   }
 
   String registerBearer(AuthResult handshakeResult) {
-    final String bearerToken = handshakeResult.getBearerToken()
+    final String bearerToken = getBearerToken()
```

Comment:
Why not create a specialized subclass of AuthResult that has getBearerToken() and pass that in to registerBearer? Where does registerBearer get called from now?

Reply:
Done, updated authentication flow to call registerBearer.